lib/snapshot/BUILD.gn

```diff
@@ -25,203 +25,146 @@ copy("generate_dart_ui") {
   ]
 }
 
-action("generate_snapshot_bin") {
-  if (is_fuchsia) {
-    snapshot_dart = "snapshot_fuchsia.dart"
-
-    # TODO(rmacnak): Fuchsia cross builds use the wrong Dart target
-    # architecture, and have added steps that depend on this error for
-    # reasonable build times (e.g., invoking the analyzer).
-    if (target_cpu == host_cpu) {
-      snapshot_kind = "core-jit"
-    } else {
-      snapshot_kind = "core"
-    }
-  } else {
-    snapshot_dart = "snapshot.dart"
-    snapshot_kind = "core"
-  }
+# Fuchsia's snapshot requires a different platform with extra dart: libraries.
+if (!is_fuchsia) {
+  compiled_action("generate_snapshot_bin") {
+    tool = "//third_party/dart/runtime/bin:gen_snapshot"
 
-  deps = [
-    ":generate_dart_ui",
```

Comment:
Dropping generate_dart_ui - was that intentional? The rule is still there, and flutter/ci/analyze.sh attempts to analyze that folder(which is no longer populated).